sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java

```diff
@@ -69,18 +72,27 @@ private static RelNode updateWindow(RelOptRuleCall call, Aggregate aggregate, Pr
       windowField = AggregateWindowFactory.getWindowFieldAt((RexCall) projNode, groupFieldIndex);
     }
 
-    BeamAggregationRel newAggregator =
-        new BeamAggregationRel(
-            aggregate.getCluster(),
-            aggregate.getTraitSet().replace(BeamLogicalConvention.INSTANCE),
-            convert(
-                aggregate.getInput(),
-                aggregate.getInput().getTraitSet().replace(BeamLogicalConvention.INSTANCE)),
-            aggregate.indicator,
-            aggregate.getGroupSet(),
-            aggregate.getGroupSets(),
-            aggregate.getAggCallList(),
-            windowField);
-    return newAggregator;
+    // rename for AggregateCalls without name
+    List<AggregateCall> renamedAggCalls = new ArrayList<>();
+
+    for (Ord<AggregateCall> ord : Ord.zip(aggregate.getAggCallList())) {
```

Comment:
I believe `aggregate.getNamedAggCalls()` works for this. But I don't think you need to call it in the converter rule. That should be in the `BeamAggregateRel` I think.

Reply:
I see. call `getNamedAggCalls()` involves with a bit refactor. Updated.